futaba/cogs/welcome/prune.py

```diff
@@ -0,0 +1,120 @@
+
+__all__ = ["Prune"]
+
+
+def prune_filter(member, prune_date, role, has_role=True):
+    """
+    Fuction used to filter members by.
+    If has_role is False it will check if the user dosen't have the role specified
+    """
+    if has_role:
+        if role in member.roles:
+            if member.joined_at < prune_date:
+                return True
+    elif role not in member.roles:
+        if member.joined_at < prune_date:
+            return True
+
+    return False
+
+
+class Prune:
+    def __init__(self, bot):
+        self.bot = bot
+        self.journal = bot.get_broadcaster("/welcome")
+
+    def setup(self):
+        pass
+
+    async def prune_member(self, ctx, days):
+        """
+        Checks if a member has had the guest role for longer that the days specified.
+        Once done will kick all members that meet that condition
+        """
+
+        # Get guild special roles
+        roles = self.bot.sql.settings.get_special_roles(ctx.guild)
+
+        # Get the date that users that join before that have to be pruned
+        prune_date = datetime.now() - timedelta(days=days)
+
+        if roles.guest:
+            logger.info(
+                "Pruning members with role %s (%d) who joined more than %d days ago",
+                roles.guest.name,
+                roles.guest.id,
+                days,
+            )
+
+            # Get users to be pruned
+            to_be_pruned = filter(
+                lambda x: prune_filter(x, prune_date, roles.guest), ctx.guild.members
+            )
+
+        else:
+            logger.info(
+                "Pruning members without role %s (%d) who joined more than %d days ago",
+                roles.member.name,
+                roles.member.id,
+                days,
+            )
+
+            # Get users to be pruned
+            to_be_pruned = filter(
+                lambda x: prune_filter(x, prune_date, roles.member, False),
+                ctx.guild.members,
+            )
+
+        pruned = []
+
+        for member in to_be_pruned:
+            try:
+                await ctx.guild.kick(
+                    member, reason=f"Pruning guests older than {days} days"
+                )
+                pruned.append(member)
+            except:
+                logger.warning("Cannnot prunt member %s (%d)", member.name, member.id)
+
+        return pruned
+
+    @commands.command(name="prune", aliases=["purge"])
+    @commands.guild_only()
+    @permissions.check_admin()
+    async def purge(self, ctx, days: int = 7):
+        """
+        Prunes users that have not used the agree command over the days specified.
+        Deafults to seven days
+        """
+
+        pruned_members = await self.prune_member(ctx, days)
+        content = f"Pruned {len(pruned_members)}"
+        embed = discord.Embed(description=content)
+        await ctx.send(embed=embed)
+
+        self.journal.send("prune", ctx.guild, content, icon="kick", cause=ctx.author)
```

Comment:
Add entry to `JOURNAL_PATHS.md` with description. Should also pass in a list of members that were pruned, maybe `pruned=pruned_members`.